filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
 func NewFilter(pattern string) (Filter, error) {
-	idx := strings.Index(pattern, WildcardChar)
+	nIdx := strings.Index(pattern, negationChar)
+	if nIdx != 0 {
```

Comment:
Hm yeah I see, somehow I misread that, guess we are not dropping those illegal chars :(